docs/integrations/consul/k8s.md

```diff
@@ -0,0 +1,313 @@
+
+Now that the ngrok Ingress Controller can communicate with the `frontend` service and `public-api` service through the Consul Service Mesh via Service Intentions, we can create an ingress to route traffic to the app. We'll create ingress objects to route traffic to the frontend service and the public-api service.
+
+:::caution Update This First!
+Update the line `host: $NGROK_DOMAIN_NAME` in the ingress object below to your ngrok domain name. For a free account, select something unique that is a subdomain of `ngrok.app`. For example, `host: my-unique-hashicups.ngrok.app`.
+:::
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-consul
+  namespace: consul
+spec:
+  ingressClassName: ngrok
+  rules:
+  - host: $NGROK_DOMAIN_NAME
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: frontend
+            port:
+              number: 3000
+      - path: /api
+        pathType: Prefix
+        backend:
+          service:
+            name: public-api
+            port:
+              number: 8080
+```
+
+:::tip This ingress object:
+- Uses the `ngrok` ingress class
+- The host is the ngrok domain name you selected that is static
+- There is a route for `/` that routes to the frontend service on port 3000
+- There is a route for `/api` that routes to the public-api service on port 8080
+:::
+
+  Open your `$NGROK_DOMAIN_NAME` domain in your browser and see the Hashicups application!
+
+  ![Hashicups App](./img/hashicups.png)
+
+
+## **Step 5**: Add OAuth Protection to the App {#add-oauth-protection-to-the-app}
+
+Now that we have the Hashicups application running, we can add OAuth protection to it. We'll use the OAuth module of the ngrok Ingress Controller to add OAuth protection to the app. This will allow us to use Google OAuth to protect the app.
+
+1. Create an [NgrokModuleSet](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/docs/user-guide/route-modules.md)
+
+  The NgrokModuleSet is a custom resource that allows you to configure the modules of the ngrok Ingress Controller. We'll create one to configure the OAuth module. Create the following `NgrokModuleSet`
+
+  ```yaml
+  kind: NgrokModuleSet
+  apiVersion: ingress.k8s.ngrok.com/v1alpha1
+  metadata:
+    name: oauth-module
+    namespace: consul
+  modules:
+    oauth:
+      google:
+        emailAddresses:
+        # Replace with your email address
+        - my-email@my-domain.com
+        # Or specify a list of domains instead of individual email addresses
+        # emailDomains:
+        # - my-domain.com
+  ```
+  :::tip Ngrok vs User Managed OAuth Application
+
+  This example uses the ngrok Managed OAuth Application for simplicity. To use your own Managed OAuth Application, you can [Creating a custom Google OAuth application](https://ngrok.com/docs/cloud-edge/modules/oauth/google/) or use an existing one.
```

Comment:
> To use your own Managed OAuth Application, you can [Creating a custom Google OAuth application]

Thanks for adding the hyperlink! I'd suggest adding the "try" here just to make this work grammatically: To use your own Managed OAuth Application, you can _try_ [Creating a custom Google OAuth application]